streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -583,6 +586,8 @@ class StreamingContext private[streaming] (
               scheduler.start()
             }
             state = StreamingContextState.ACTIVE
+            startTime = System.currentTimeMillis()
```

Comment:
Setting the start here is a change from the previous pr where it's set in the constructor. It seems fine, but what is your reasoning behind the change.